File: src/main/java/com/example/jobs/dto/SearchAnalyticsResponse.java
package com.example.jobs.dto;

import com.fasterxml.jackson.annotation.JsonProperty;
import java.util.ArrayList;
import java.util.List;
import lombok.Getter;

@Getter
public class SearchAnalyticsResponse {
  private Integer days;
  private List<EndPointStat> endPoints = new ArrayList<>();

  @JsonProperty("top_queries")
  private List<QueryStat> topQueries = new ArrayList<>();

  @JsonProperty("top_searched_tags")
  private List<TagStat> tags = new ArrayList<>();

  @JsonProperty("top_user_agents")
  private List<UserAgentStat> topUserAgents = new ArrayList<>();

  private Long totalRequests = 0L;
  private Long zeroResultSearches = 0L;

  public SearchAnalyticsResponse(Integer days) {
    this.days = days;
  }

  public void addEndpoint(EndPointStat e) { this.endPoints.add(e); }
  public void addQuery(QueryStat q) { this.topQueries.add(q); }
  public void addTag(TagStat t) { this.tags.add(t); }
  public void setCounters(long total, long zero) {
    this.totalRequests = total;
    this.zeroResultSearches = zero;
  }
}

File: src/main/kotlin/com/example/jobs/dto/SearchAnalyticsResponse.kt
package com.example.jobs.dto

import com.fasterxml.jackson.annotation.JsonIgnoreProperties
import com.fasterxml.jackson.annotation.JsonProperty

@JsonIgnoreProperties(ignoreUnknown = true)
data class SearchAnalyticsResponse(
    val days: Int,
    val endpoints: List<EndpointStat>,

    @field:JsonProperty("top_queries")
    val topQueries: List<QueryStat>,

    @field:JsonProperty("top_searched_tags")
    val topSearchedTags: List<TagStat>,

    @field:JsonProperty("top_user_agents")
    val topUserAgents: List<UserAgentStat> = emptyList(), // Defaulted to safe empty list

    @field:JsonProperty("total_requests")
    val totalRequests: Long,

    @field:JsonProperty("zero_result_searches")
    val zeroResultSearches: Long
)

data class EndpointStat(
    val endpoint: String,
    val count: Long,
    @field:JsonProperty("avg_results") val avgResults: Long
)

data class QueryStat(
    val query: String,
    val count: Long,
    @field:JsonProperty("avg_results") val avgResults: Long
)

data class TagStat(
    val tag: String,
    val count: Long
)

data class UserAgentStat(
    @field:JsonProperty("user_agent") val userAgent: String,
    val count: Long
)